src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -32,14 +32,15 @@
 #include <nupic/algorithms/SpatialPooler.hpp>
 #include <nupic/math/Math.hpp>
 #include <nupic/math/Topology.hpp>
-#include <nupic/proto/SpatialPoolerProto.capnp.h>
+
+#define VERSION 2  // version for stream serialization
 
 using namespace std;
 using namespace nupic;
 using namespace nupic::algorithms::spatial_pooler;
 using namespace nupic::math::topology;
 
-static const Real PERMANENCE_EPSILON = 0.000001;
+static const Real PERMANENCE_EPSILON = 0.000001f;
```

Comment:
this is a second time I see the PERMANENCE_EPSILON defined, also, possibly with different (?) value. Should be unified into some class (or import from there, eg nupic::PERMANENCE_EPSILON ?)